public/app/core/components/OptionsUI/units.tsx

```diff
@@ -1,11 +1,35 @@
+import { css } from '@emotion/css';
 import React from 'react';
 
-import { FieldConfigEditorProps, UnitFieldConfigSettings } from '@grafana/data';
-import { UnitPicker } from '@grafana/ui';
+import { FieldConfigEditorProps, GrafanaTheme2, UnitFieldConfigSettings } from '@grafana/data';
+import { IconButton, UnitPicker, useStyles2 } from '@grafana/ui';
 
-export const UnitValueEditor: React.FC<FieldConfigEditorProps<string, UnitFieldConfigSettings>> = ({
-  value,
-  onChange,
-}) => {
+type Props = FieldConfigEditorProps<string, UnitFieldConfigSettings>;
+
+export function UnitValueEditor({ value, onChange, item }: Props) {
+  const styles = useStyles2(getStyles);
+  if (item?.settings?.isClearable && value != null) {
```

Comment:
Do types elsewhere need to be updated? `value` here is typed to just `string`, yet it looks like you're expecting it to be also null? This can lead to problems with devs incorrectly assuming this value can never not be a string?

Reply:
`value` is typed as `value?: string`, so this *could* be `value !== undefined` but I always find that much less reliable the `!= null` since it works for for null also

Reply:
hmmmm not sure if im seeing that, but i trust you